Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,284 @@
+      CONFIGURATOR_PROXY,
+      [...mockArgs],
+    );
+
+    const metadata1: { [key: string]: any } = {
+      SET_GOVERNOR: {
+        Old_value: createAddress("0x123"),
+        New_value: createAddress("0x234"),
+      },
+    };
+    const metadata = {
+      "Collateral Asset - Collateral-1":
+        "0x0000000000000000000000000000000000001247",
+      "Collateral Asset - Collateral-2":
+        "0x0000000000000000000000000000000000012567",
+      "Collateral Asset - Collateral-3":
+        "0x0000000000000000000000000000000000001289",
+      "Collateral Asset - Collateral-4":
+        "0x0000000000000000000000000000000000001290",
+    };
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(1);
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: `Change of asset values due to governance proposal`,
+        description: `The asset ${mockAssetTokenAddress} has been modified by a governance proposal`,
+        alertId: "GOV-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: txEvent.network.toString(),
+        metadata: {
+          ...metadata1,
+          ...metadata,
+        },
+      }),
+    ]);
+
+    mockProvider.clear();
+    expect(findings.length).toStrictEqual(1);
+  });
+
+  it("should return 0 findings for event other than proposal change", async () => {
+    mockProvider.setNetwork(1);
+    txEvent.addEventLog(
+      "event mockEvent(address value1, uint8 value2)",
+      CONFIGURATOR_PROXY,
+      [createAddress("0x9467"), 2],
+    );
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(0);
+  });
+
+  it("should return 0 findings for multiple non-governance events", async () => {
+    mockProvider.setNetwork(1);
+    const nonGovEvents = [
+      "event Transfer(address indexed from, address indexed to, uint256 value)",
+      "event Approval(address indexed owner, address indexed spender, uint256 value)",
+      "event Deposit(address indexed user, address reciepient, uint256 amount)",
+    ];
+
+    nonGovEvents.forEach((eventSignature) => {
+      txEvent.addEventLog(eventSignature, createAddress("0xABCDEF"), [
+        createAddress("0x12345"),
+        createAddress("0x67890"),
+        1000,
+      ]);
+    });
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(0);
+  });
+
+  it("should a finding for a governance event out of multiple non-governance events", async () => {
+    mockProvider.setNetwork(1);
+    const nonGovEvents = [
+      "event Transfer(address indexed from, address indexed to, uint256 value)",
+      "event Approval(address indexed owner, address indexed spender, uint256 value)",
+      "event Deposit(address indexed user, address reciepient, uint256 amount)",
+    ];
+    nonGovEvents.forEach((eventSignature) => {
+      txEvent.addEventLog(eventSignature, createAddress("0xABCDEF"), [
```

Comment:
Same comment regarding the second argument to `.addEventLog` being the correct address.